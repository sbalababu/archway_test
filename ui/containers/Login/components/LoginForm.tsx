import * as React from 'react';
import { Button, Form, Icon, Input } from 'antd';
import { InjectedFormikProps, withFormik } from 'formik';
import { Login } from '../../../models/Login';
import { SPNEGO } from '../../../constants';

interface Props {
  onSubmit: (payload: Login) => void;
  loggingIn: boolean;
  error: string;
  authType: string;
}

class LoginForm extends React.PureComponent<InjectedFormikProps<Props, Login>> {
  public render() {
    const {
      handleSubmit,
      handleChange,
      values: { username, password },
      authType,
    } = this.props;

    return (
      <Form onSubmit={handleSubmit} className="LoginForm" layout="vertical">
        {authType === SPNEGO || (
          <React.Fragment>
            <Form.Item>
              <Input
                name="username"
                onChange={handleChange}
                size="large"
                value={username}
                prefix={<Icon type="user" />}
                placeholder="username"
              />
            </Form.Item>
            <Form.Item>
              <Input
                name="password"
                onChange={handleChange}
                size="large"
                value={password}
                prefix={<Icon type="lock" />}
                placeholder="password"
                type="password"
              />
            </Form.Item>
          </React.Fragment>
        )}
        <Form.Item>
          <Button type="primary" size="large" htmlType="submit">
            Log In
          </Button>
        </Form.Item>
      </Form>
    );
  }
}

const LoginFormRender = withFormik<Props, Login>({
  handleSubmit: (values, { props }) => props.onSubmit(values),
})(LoginForm);

export default LoginFormRender;
