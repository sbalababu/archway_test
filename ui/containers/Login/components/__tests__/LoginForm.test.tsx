import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import LoginForm from '../LoginForm';

describe('LoginForm', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<LoginForm onSubmit={() => null} loggingIn={false} error="" authType="" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
