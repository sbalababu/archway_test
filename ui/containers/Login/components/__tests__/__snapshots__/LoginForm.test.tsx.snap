// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders correctly 1`] = `
<Formik
  authType=""
  enableReinitialize={false}
  error=""
  handleSubmit={[Function]}
  initialValues={
    Object {
      "authType": "",
      "error": "",
      "loggingIn": false,
    }
  }
  isInitialValid={false}
  loggingIn={false}
  onSubmit={[Function]}
  render={[Function]}
  validateOnBlur={true}
  validateOnChange={true}
/>
`;
